brainglobe_stitch/image_mosaic.py

```diff
@@ -340,3 +346,366 @@ def read_big_stitcher_transforms(self) -> None:
+
+            channel_ds_list.append(ds_list)
+
+        # Place the tiles in reverse order of acquisition
+        for tile in self.tiles[-1::-1]:
+            current_tile_data = tile.data_pyramid[0].compute()
+            channel_ds_list[tile.channel_id][0][
+                tile.position[0] : tile.position[0] + z_size,
+                tile.position[1] : tile.position[1] + y_size,
+                tile.position[2] : tile.position[2] + x_size,
+            ] = current_tile_data
+
+            # Use a simple downsample for the other resolutions
+            for i in range(1, len(resolutions)):
+                scaled_position = tile.position // resolutions[i, -1::-1]
+                scaled_size = (
+                    (z_size + (resolutions[i][2] > 1)) // resolutions[i][2],
+                    (y_size + (resolutions[i][1] > 1)) // resolutions[i][1],
+                    (x_size + (resolutions[i][0] > 1)) // resolutions[i][0],
+                )
+                channel_ds_list[tile.channel_id][i][
+                    scaled_position[0] : scaled_position[0] + scaled_size[0],
+                    scaled_position[1] : scaled_position[1] + scaled_size[1],
+                    scaled_position[2] : scaled_position[2] + scaled_size[2],
+                ] = current_tile_data[
+                    :: resolutions[i][2],
+                    :: resolutions[i][1],
+                    :: resolutions[i][0],
+                ]
+
+            print(f"Done tile {tile.id}")
+
+        assert self.xml_path is not None
+
+        write_bdv_xml(
+            output_path.with_suffix(".xml"),
+            self.xml_path,
+            output_path,
+            fused_image_shape,
+            self.num_channels,
+        )
+
+        output_file.close()
+
+    def get_metadata_for_zarr(
+        self,
+        pyramid_depth: int = 6,
+        downscale_factors: Tuple[int, int, int] = (1, 2, 2),
+    ) -> Tuple[List[List[Dict]], List[Dict]]:
+        """
+        Prepare the metadata for a zarr file.
+
+        Parameters
+        ----------
+        pyramid_depth: int
+            The depth of the resolution pyramid.
+        downscale_factors: Tuple[int, int, int]
+            The factors to downscale the image by in the z, y, x dimensions.
+
+        Returns
+        -------
+        Tuple[List[List[Dict]], List[Dict]]
+            A tuple with the coordinate transformations and axes metadata.
+        """
+        axes = [
+            {"name": "z", "type": "space", "unit": "micrometer"},
+            {"name": "y", "type": "space", "unit": "micrometer"},
+            {"name": "x", "type": "space", "unit": "micrometer"},
+        ]
+
+        coordinate_transformations = []
+        for i in range(pyramid_depth):
+            coordinate_transformations.append(
+                [
+                    {
+                        "type": "scale",
+                        "scale": [
+                            self.z_resolution * downscale_factors[0] ** i,
```

Comment:
I tried extracting it, but it felt a tad awkward since above I just need the factors, whereas here I'm calculating the specific resolution in um. It felt like I more code than I removed! I'll keep this in mind in the future if I reuse this again somewhere.

Reply:
makes sense!